File: scripts/archive/assign_colors.py
import random
import time
from supabase import create_client, Client

# Configuration
url: str = "YOUR_SUPABASE_URL"
key: str = "YOUR_SUPABASE_ANON_KEY"

COLORS = ['#2563eb', '#06b6d4', '#7c3aed', '#db2777', '#059669', '#d97706', '#4f46e5', '#be123c']

def main():
    print("Connecting to Smart Tutoring Monitor Database...")
    supabase: Client = create_client(url, key)

    print("Fetching students...")
    response = supabase.table("students").select("*").execute()
    students = response.data

    if not students:
        print("No students found.")
        return

    print(f"Processing {len(students)} students...")
    updated_count = 0

    for student in students:
        # Check if color is missing or valid
        current_color = student.get('color')
        
        # If no color, or color not in our palette (optional, but good for consistency)
        if not current_color or current_color not in COLORS:
            new_color = random.choice(COLORS)
            print(f"Assigning color {new_color} to {student.get('name')}...")
            
            try:
                supabase.table("students").update({"color": new_color}).eq("id", student['id']).execute()
                updated_count += 1
                time.sleep(0.1) # Rate limit
            except Exception as e:
                print(f"Error updating student {student.get('id')}: {e}")

    print(f"\n[SUCCESS] Updated {updated_count} student records with colors.")

if __name__ == "__main__":
    main()

File: scripts/archive/assign_colors.js
import { createClient } from '@supabase/supabase-js';

// Configuration
const url = "YOUR_SUPABASE_URL";
const key = "YOUR_SUPABASE_ANON_KEY";

const COLORS = ['#2563eb', '#06b6d4', '#7c3aed', '#db2777', '#059669', '#d97706', '#4f46e5', '#be123c'];

const supabase = createClient(url, key);

async function main() {
    console.log("Connecting to Smart Tutoring Monitor Database (Node.js)...");

    console.log("Fetching students...");
    const { data: students, error } = await supabase.from('students').select('*');

    if (error) {
        console.error("Error fetching students:", error);
        return;
    }

    if (!students || students.length === 0) {
        console.log("No students found.");
        return;
    }

    console.log(`Processing ${students.length} students...`);
    let updatedCount = 0;

    for (const student of students) {
        const currentColor = student.color;

        // If missing or not in palette (optional check)
        if (!currentColor || !COLORS.includes(currentColor)) {
            const newColor = COLORS[Math.floor(Math.random() * COLORS.length)];
            console.log(`Assigning color ${newColor} to ${student.name}...`);

            const { error: updateError } = await supabase
                .from('students')
                .update({ color: newColor })
                .eq('id', student.id);

            if (updateError) {
                console.error(`Error updating student ${student.id}:`, updateError);
            } else {
                updatedCount++;
            }
        }
    }

    console.log(`\n[SUCCESS] Updated ${updatedCount} student records with colors.`);
}

main();
